tests/providers/aws/aws_cli.py

```diff
@@ -0,0 +1,167 @@
+# -*- coding: utf-8 -*-
+# Copyright 2021 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Call libcloudforensics CLI tool for e2e testing."""
+import subprocess
+from typing import List, Optional
+
+from libcloudforensics import logging_utils
+from libcloudforensics.errors import ResourceCreationError
+from libcloudforensics.errors import ResourceNotFoundError
+from libcloudforensics.providers.aws.internal import ebs
+from libcloudforensics.providers.aws.internal import ec2
+from libcloudforensics.providers.aws.internal.account import AWSAccount
+
+logging_utils.SetUpLogger(__name__)
+logger = logging_utils.GetLogger(__name__)
+
+
+class AWSCLI:
+  """AWSCLI calls the libcloudforensics CLI tool for AWS operations."""
+  def __init__(self, aws: AWSAccount) -> None:
+    """Initialize the CLI class.
+
+    Attributes:
+      aws (AWSAccount): The AWS account to work with.
+    """
+    self.aws = aws
+
+  def StartAnalysisVm(
+      self,
+      vm_name: str,
+      zone: str,
+      attach_volumes: Optional[List[str]] = None) -> ec2.AWSInstance:
+    """Start an analysis VM.
+
+    Args:
+      vm_name (str): The name of the instance to start.
+      zone (str): The zone in which to start the instance.
+      attach_volumes (List[str]): Optional. List of volume names to attach to
+          the VM.
+
+    Returns:
+      ec2.AWSInstance: A ec2.AWSInstance object that represents the
+          started VM.
+
+    Raises:
+      ResourceCreationError: If the VM could not be created.
+      ResourceNotFoundError: If the created VM could not be found.
+    """
+    cmd = 'cloudforensics aws {0:s} startvm {1:s}'.format(  # pylint: disable=line-too-long
+        zone, vm_name)
+    if attach_volumes:
+      cmd += ' --attach_volumes={0:s}'.format(','.join(attach_volumes))
+    logger.info('CLI command: {0:s}'.format(cmd))
+    try:
+      output = subprocess.check_output(
+          cmd.split(), stderr=subprocess.STDOUT, shell=False)
+      logger.info(output)
+      try:
+        return self.aws.ec2.GetInstancesByName(vm_name)[0]
+      except ResourceNotFoundError as error:
+        raise ResourceNotFoundError(
+            'Failed finding created VM in account profile {0:s}'.format(
+                self.aws.aws_profile), __name__) from error
```

Comment:
Can we move this to outside of the wrapping `try` block? It seems unlikely that this part would ever raise a `CalledProcessError` exception.

Also - should this part go in the actual test?

Reply:
Moved the block.

Re: `Also - should this part go in the actual test?` Great question. I tried to make the actual e2e CLI file as close as possible as the Python one, hence why this code is here. The other option is to return whatever we need from the output of the CLI call (mostly, instance + disk names as strings) and then grab the instance / disk as a Python object in the test directly. I don't have a strong preference for either option, so up to you.